File: amazia-console/app/Notice/Controller/GetNoticeCategoriesController.php
<?php

namespace App\Notice\Controller;

use App\Http\Controllers\Controller;
use App\Notice\Service\GetNoticeCategoriesService;

/**
 * フェーズ19: お知らせ分類マスタ Controller（Console Pass-through）。
 * GET /api/notice-categories
 */
class GetNoticeCategoriesController extends Controller
{
    private GetNoticeCategoriesService $service;

    public function __construct(GetNoticeCategoriesService $service)
    {
        $this->service = $service;
    }

    public function __invoke()
    {
        $response = $this->service->findAll();
        return response()->json($response->json(), $response->status());
    }
}

File: amazia-core/src/main/java/com/example/notice/controller/GetNoticeCategoriesController.java
package com.example.notice.controller;

import com.example.notice.dto.NoticeCategoryDto;
import com.example.notice.service.GetNoticeCategoriesService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * お知らせ分類マスタ取得（GET /api/notice-categories）。認証不要。
 */
@RestController
@RequestMapping("/api/notice-categories")
public class GetNoticeCategoriesController {

    private final GetNoticeCategoriesService service;

    public GetNoticeCategoriesController(GetNoticeCategoriesService service) {
        this.service = service;
    }

    @GetMapping
    public List<NoticeCategoryDto> getAll() {
        return service.findAll();
    }
}
